File: src/1275-Find-Winner-on-a-Tic-Tac-Toe-Game/1275.java
class Solution {
    public String tictactoe(int[][] moves) {
        int[][] score = new int[2][8];
        int p = 0;
        for (int[] it : moves) {
            score[p][it[0]]++; score[p][it[1] + 3]++;
            score[p][6] += (it[0] == it[1] ? 1 : 0);
            score[p][7] += (it[0] + it[1] == 2 ? 1 : 0);
            for (int x : score[p])
                if (x == 3) return p == 0 ? "A" : "B";
            p ^= 1;
        }
        return moves.length == 9 ? "Draw" : "Pending";
    }
}

File: src/1275-Find-Winner-on-a-Tic-Tac-Toe-Game/1275.py
class Solution:
    def tictactoe(self, moves: List[List[int]]) -> str:
        score, p = [[0]*8 for _ in range(2)], 0

        for i, j in moves:
            score[p][i] += 1
            score[p][3+j] += 1
            score[p][6] += (i == j)
            score[p][7] += (i+j == 2)
            if any(x == 3 for x in score[p]): 
                return ["A", "B"][p]
            p ^= 1 
            
        return "Draw" if len(moves) == 9 else "Pending"